datatypes/mcs_datatype.cpp

```diff
@@ -478,49 +478,49 @@ int TypeHandlerVarbinary::storeValueToField(rowgroup::Row& row, int pos, StoreFi
 int TypeHandlerSInt64::storeValueToField(rowgroup::Row& row, int pos, StoreField* f) const
 {
   int64_t val = row.getIntField<8>(pos);
-  return f->store_xlonglong(val);
+  return f->store_longlong(val);
 }
 
 int TypeHandlerUInt64::storeValueToField(rowgroup::Row& row, int pos, StoreField* f) const
 {
   uint64_t val = row.getUintField<8>(pos);
-  return f->store_xlonglong(static_cast<int64_t>(val));
+  return f->store_ulonglong(static_cast<int64_t>(val));
 }
 
 int TypeHandlerInt::storeValueToFieldSInt32(rowgroup::Row& row, int pos, StoreField* f) const
 {
   int64_t val = row.getIntField<4>(pos);
-  return f->store_xlonglong(val);
+  return f->store_longlong(val);
 }
 
 int TypeHandlerInt::storeValueToFieldUInt32(rowgroup::Row& row, int pos, StoreField* f) const
 {
   uint64_t val = row.getUintField<4>(pos);
-  return f->store_xlonglong(static_cast<int64_t>(val));
+  return f->store_ulonglong(static_cast<int64_t>(val));
```

Comment:
There is no difference b/w U64 and U32 - same Field::store_ulonglong() call ?

Reply:
There is a difference between ulonglong and longlong. Even though we send field a signed, in the underlying field_store(), we need to pass the unsigned flag differently. The way the flag had been calculated was based on something that may or may not be true.

Reply:
We agreed that it is better to have the cast inside the store_ulonglong() function. 
Function is now store_ulonglong(uint64_t val);